### ISSUE TITLE
Missing comma in function arguments can result in a misleading diagnostic about missing arguments

### ISSUE THREAD

**Issue by nagisa** [rust-lang/rust#120678]
### Code

```Rust
fn a(a: u64, b: u64, c: u64) { todo!() }

fn main() {
    let a_neat_first_argument = 42;
    let a_neat_second_argument_array_in_disguise = [42u8; 3];
    a(
        a_neat_first_argument,
        a_neat_second_argument_array_in_disguise.len() as u64
        &a_neat_second_argument_array_in_disguise as *const _ as u64,
    )
}
```

### Current output

```Shell
error[E0061]: this function takes 3 arguments but 2 arguments were supplied
  --> src/main.rs:6:5
   |
6  |       a(
   |  _____^-
7  | |         a_neat_first_argument,
8  | |         a_neat_second_argument_array_in_disguise.len() as u64
9  | |         &a_neat_second_argument_array_in_disguise as *const _ as u64,
10 | |     )
   | |_____- an argument of type `u64` is missing
   |
note: function defined here
  --> src/main.rs:1:4
   |
1  | fn a(a: u64, b: u64, c: u64) { todo!() }
   |    ^ ------  ------  ------
help: provide the argument
   |
6  ~     a(a_neat_first_argument, a_neat_second_argument_array_in_disguise.len() as u64
7  +         &a_neat_second_argument_array_in_disguise as *const _ as u64, /* u64 */)
   |
```

### Desired output

```Shell
Admittedly a long shot, but a suggestion to add a comma after `a_neat_second_argument_array_in_disguise.len() as u64` would be neat if we were able to somehow make sense out of this.
```

### Rationale and extra context

This is pretty hard one to do something about – the second argument is actually a valid expression even with the missing comma. However the layout of the code is screaming "TWO SEPARATE ARGUMENTS HERE" no matter how you look at this. Not to mention the missing comma can be pretty hard to spot, so the user may be left counting the arguments and trying to understand how in the world Rust counts 2 rather than 3 here.

### Rust Version

```Shell
1.77.0-nightly (2024-02-01 bf3c6c5bed498f41ad81)
```

**Comment by fmease** [comment]
We could hackily check in the code responsible for the arg count mismatch diagnostic if there are arguments that span multiple lines with `Span`/`SourceMap` and are of a certain expression kind (`BinOp` (but not surrounded by parens), …) and suggest inserting a comma.

**Comment by fmease** [comment]
Of course, `rustfmt` on save would've prevented this confusion.
